tt_eager/tt_dnn/op_library/softmax/kernels/compute/softmax.cpp

```diff
@@ -132,34 +133,50 @@ void MAIN {
             pack_reconfig_data_format(cb_exps);
             copy_tile_to_dst_init_short(); // need to copy from CB to DST to be able to run sfpu math
             exp_tile_init<EXP_APPROX>();
-            for (uint32_t wt = 0; wt < Wt; wt+=ndst) {
+            if (mask_padded_data) {
```

Comment:
More of a code cleanliness/organization thing, but maybe we should just take what you had before, and put it into an inline template function with the template being a bool for mask padding. That way we aren't duplicating common code. Not necessary for this pr though since priority is to just get the fix in for ci

Then here we just have
```
if (mask_padded_data) {
    templated_fn<true>(...);
} else {
    templated_fn<false>(...);
}

Reply:
yeah I can get a PR for this after the fix